search/src/Components/wrapper/datasets.js

```diff
@@ -49,22 +68,53 @@ class DatasetComponent extends Component {
         })
     }
 
+    handleSwitchEditingMode () {
+        this.setState({
+            editing: !this.state.editing
+        })
+    }
+
     render() {
+        const editingToggle = 
+        this.state.editing_id_list.length > 0 && this.state.upload_id === "*"
+        ?
+        (
+            <FormControlLabel
+            control={
+            <Switch
+                checked={this.state.editing}
+                onClick={this.handleSwitchEditingMode}
+                name="checkedA"
+                color="primary"
+            />
+            }
+            label="Show only Datasets I own"
```

Comment:
Can I understand in this way:
We don't need the toggle. The backend just return back the whole list with a flag of whether it's editable for this user. Editable and non-editable should just be put into different tables but in a single page.

Reply:
Make the backend work, and then there will be lots of possibilities in the frontend:
* the current display, but with an "edit" button where permitted
* a toggle like proposed
* a combination of the above
* separate tables

Reply:
I can make a flag for editable in the dataset request for now, but I do think that separating them makes future work on editing draft more easily.

Reply:
Not sure what you mean here. I don't think the frontend needs to query for editable datasets only, so the change is in the response not the request. When you say separating them makes future work easier, do you mean that the UX is easier if the user can see only the datasets they uploaded, or that they can edit? Yes, that may be true, but it can be added with a filter in the frontend, and later the backend.